From b4c0ba6e13e0389911c08c7b35e8c716972d1752 Mon Sep 17 00:00:00 2001
From: Tim-cy Yang <Tim-cy.Yang@mediatek.com>
Date: Tue, 4 Nov 2025 15:15:26 +0800
Subject: [PATCH 20/23] board: mediatek: common: add FIT anti-rollback check
 before upgrading

[Description]
Add FIT anti-rollback check before upgrading. Check the anti-rollback
version for all configuration nodes.

[Release-Log]
N/A

Change-Id: Ib51f6e090a730bc25f75b15a9bf47c176cbf7c10
Signed-off-by: Tim-cy Yang <Tim-cy.Yang@mediatek.com>
CR-Id: WCNCR00402800
---
 board/mediatek/common/verify_helper.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/board/mediatek/common/verify_helper.c b/board/mediatek/common/verify_helper.c
index 596bd7794ae..9ea5db4e2e3 100644
--- a/board/mediatek/common/verify_helper.c
+++ b/board/mediatek/common/verify_helper.c
@@ -16,6 +16,7 @@
 #include "upgrade_helper.h"
 #include "verify_helper.h"
 #include "untar.h"
+#include "mtk_ar.h"
 
 /* Only supports native byte-order for squashfs */
 #define SQUASHFS_MAGIC		0x73717368
@@ -138,6 +139,10 @@ static int fit_all_config_verify(const void *fit)
 
 			if (fit_config_verify(fit, noffset))
 				return 0;
+#ifdef CONFIG_MTK_ANTI_ROLLBACK
+			if (fit_config_ar_ver_verify(fit, noffset, NULL))
+				return 0;
+#endif
 			printf("\n");
 		}
 	}
-- 
2.45.2

